Mathlib/Data/Matroid/Rank/Cardinal.lean

```diff
@@ -342,4 +356,34 @@ instance invariantCardinalRank_comap (M : Matroid β) [InvariantCardinalRank M]
 
 end Instances
 
+theorem finiteRk_iff_cRank_lt_aleph0 : M.FiniteRk ↔ M.cRank < ℵ₀ := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ⟨?_⟩⟩
+  · have ⟨B, hB, fin⟩ := h
+    exact hB.cardinalMk_eq_cRank ▸ lt_aleph0_iff_finite.mpr fin
+  have ⟨B, hB⟩ := M.exists_base
+  simp_rw [← finite_coe_iff, ← lt_aleph0_iff_finite]
+  exact ⟨B, hB, (le_ciSup_of_le (bddAbove_range _) ⟨B, hB⟩ le_rfl).trans_lt h⟩
+
+theorem Indep.base_of_cRank_le [M.FiniteRk] (ind : M.Indep I) (le : M.cRank ≤ #I) : M.Base I :=
+  ind.base_of_maximal fun _J ind_J hIJ ↦ ind.finite.eq_of_subset_of_encard_le' hIJ <|
+    toENat.monotone' <| ind_J.cardinalMk_le_cRank.trans le
+
+theorem Spanning.base_of_le_cRank [M.FiniteRk] (h : M.Spanning X) (le : #X ≤ M.cRank) :
+    M.Base X := by
+  have ⟨B, hB, hBX⟩ := h.exists_base_subset
+  rwa [← hB.finite.eq_of_subset_of_encard_le' hBX
+    (toENat.monotone' <| le.trans (hB.cardinalMk_eq_cRank).ge)]
+
+theorem Indep.base_of_cRank_le_of_finite (ind : M.Indep I)
```

Comment:
Isn't this the older version of this section before these were golfed?

Reply:
Yes, this diff will disappear once #21512 is merged into master and master is merged into this PR.